### PR TITLE
Switched links to be indexed by rel

### LINKED ISSUE #13: Configure whether links/relations are index by rel
The configuration would look like this:

``` yaml
fsc_hateoas:
    json:
        index_relations_by_rel: true
        index_links_by_rel: true
```

### PR DESCRIPTION
**Pull request by baldurrensch**
This fixes #13

Still needs another test to test multiple links. And I will fix the one failing test.

### CODE REVIEW

**Comment by baldurrensch** [comment]
Requires https://github.com/TheFootballSocialClub/FSCHateoasBundle/pull/15 to be merged in. Waiting for the PR to be merged in. Then I will rebase.

**Comment by baldurrensch** [comment]
Rebased. Failing test fixed. I will add another unit test tomorrow. Any other suggestions?

File: Serializer/LinkSerializationHelper.php [hunk 1]
```diff
@@ -32,6 +32,27 @@ public function addLinksToXMLSerialization(array $links, XmlSerializationVisitor
 
     public function createGenericLinksData(array $links, GenericSerializationVisitor $visitor)
     {
-        return $visitor->getNavigator()->accept($links, $this->typeParser->parse('array<FSC\HateoasBundle\Model\Link>'), $visitor);
+        $result = array();
```
Comment (adrienbrault) [inline Serializer/LinkSerializationHelper.php]: `$result` => `$serializedLinks` ?

File: Serializer/LinkSerializationHelper.php [hunk 2]
```diff
@@ -32,6 +32,27 @@ public function addLinksToXMLSerialization(array $links, XmlSerializationVisitor
...
+
+        foreach ($links as $link) {
+            $rel = $link->getRel();
+            $link->setRel(null);    // To avoid serialization
+
+            $serializedLink = $visitor->getNavigator()->accept($link, $this->typeParser->parse('FSC\HateoasBundle\Model\Link'), $visitor);
+            if (!empty($result[$rel])) {
```
Comment (adrienbrault) [inline Serializer/LinkSerializationHelper.php]: Could you use `isset` instead of `!empty` ?

**Comment by baldurrensch** [comment]
@adrienbrault Any other suggestions?

**Comment by adrienbrault** [comment]
Hum, is it possible for a relation to be embedded 2 times with json-hal ?

**Comment by baldurrensch** [comment]
If you mean, whether you can have the same resource (type), with a specific rel, then yes, you can have them embedded multiple times. Consider the following example, in this case the `orders` relation is embedded twice, but they are two different resources (i.e. order with id 123, and order with id 124). However, embedding the same resource twice (say with id 123), seems legit to me, but does not make much sense. 

``` json
{
  "_links": {
    "self": { "href": "/orders" },
    "next": { "href": "/orders?page=2" },
    "find": { "href": "/orders{?id}", "templated": true },
    "admin": [
      { "href": "/admins/2", "title": "Fred" },
      { "href": "/admins/5", "title": "Kate" }
    ]
  },
  currentlyProcessing: 14,
  shippedToday: 20,
  "_embedded": {
   "orders": [{
       "_links": {
         "self": { "href": "/orders/123" },
         "basket": { "href": "/baskets/98712" },
         "customer": { "href": "/customers/7809" }
       },
       "total": 30.00,
       "currency": "USD",
       "status": "shipped",
     },{
       "_links": {
         "self": { "href": "/orders/124" },
         "basket": { "href": "/baskets/97213" },
         "customer": { "href": "/customers/12369" }
       },
       "total": 20.00,
       "currency": "USD",
       "status": "processing"
    }]
  }
}
```

**Comment by adrienbrault** [comment]
Thanks

**Pull request merged**